### PR TITLE
BFD-1637: Add tests for loading multiple years for same bene id

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-1637](https://jira.cms.gov/browse/BFD-1637)

**User Story or Bug Summary:**

As a BFD engineer, I want there to be test coverage for processing UPDATEs with multiple reference years for the same beneficiary from the same file, so that I can confidently process synthetic beneficiaries with multiple reference years because this is not how non-synthetic data currently behaves.

---

### What Does This PR Do?

- Added tests for testing a single bene with multiple enrollment years in the same file, for both single and multiple batches
- Adjusted the batch size to be loaded from the load config instead of hardcoded (it still gets passed the old default value in the app config)
    - This was to allow for tests to adjust the batch size
- Added a file to support the test
- Added a method to support checking bene monthly table for any given year for a bene

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.
* Ensure the test asserts the AC of the story: After benes are processed, the beneMonthly fields are populated for each year that there is an UPDATE and any additional HICNs ad MBIs are also populated

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java [hunk 1]
```diff
@@ -93,6 +94,99 @@ public void loadSampleA() {
...
+      assertEquals(60, beneficiaryFromDb.getBeneficiaryMonthlys().size());
+      // Checks every month in the beneficiary monthly table for each of the years loaded
+      assertBeneficiaryMonthly(beneficiaryFromDb, 1990);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2000);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2001);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2003);
+      assertBeneficiaryMonthly(beneficiaryFromDb, 2019);
+
+      // Set the batch to 10 so all 5 records are in the same batch
+      LoadAppOptions options = CcwRifLoadTestUtils.getLoadOptionsWithBatchSize(10);
```
Comment (christrotteradhoc) [inline apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:174]: Isn't this test supposed to do more than 1 batch so the batch size should be smaller than 5? Also, how can this influence the batch size here, after the records were already loaded?

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-multiple-entries-same-bene.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+INSERT|567834|MO|123|12345|17-MAR-1981|1|1|1|1|N|20|0|0|543217066U|Doe|John|A|3456789|17-Mar-1981|1990|4|5|6|7|8|A|B|C|D|4|19631003|5|AA|AA|AA|AA|AA|AA|AA|AA|AA|AA|AA|AA|4|4|AA|AA|AA|AA|AA|AA|AA|AA|AA|AA|AA|AA|5|C|C|C|C|C|C|C|C|C|C|C|C|1|1|1|1|1|1|1|1|1|1|0|0|H1225|H1225|H1225|H1225|H1225|H1225|H1225|H1225|H1225|H1225|0|0|003|003|003|003|003|003|003|003|003|003|||00|00|00|00|00|00|00|00|00|00|||S4607|S4607|S4607|S4607|S4607|S4607|S4607|S4607|S4607|S4607|S4607|0|003|003|003|003|003|003|003|003|003|003|003||000|000|000|000|000|000|000|000|000|000|000||Y|Y|Y|Y|Y|Y|Y|Y|Y|Y|Y|N|**|**|**|**|**|**|**|**|**|**|**|**|00|00|01|01|05|06|07|08|01|02|00|00|204 SOUTH ST|7560 123TH ST|SURREY|DAEJEON SI 34867|COLOMBIA|SURREY|PODUNK|IA|123456789|30-JUL-2020||1
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-multiple-entries-same-bene.txt:2]: I think the use case is all UPDATEs as we don't expect INSERTs and UPDATEs spread across multiple batches to necessarily be handled.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java (hunk 1 above)
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:174]: Good catch! I copy pasted and didn't change the value.

Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java]: Fixed this! Options is now properly passed into loadSample so it gets used, and the batch test's value is set to 2.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-multiple-entries-same-bene.txt (hunk 2 above)
Comment (lsmitchell) [reply apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-multiple-entries-same-bene.txt]: I added that first insert so the bene existed to update, but given your comment I adjusted the tests to have an initial load (with no ref year so it didnt populate bene history) and then 6 updates in the file.

**Review by christrotteradhoc — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]
LGTM.

**Pull request merged**